src/types/complex-types/model.ts

```diff
@@ -458,7 +468,7 @@ export class ModelType<
     return this.cloneAndEnhance({ properties })
   }
 
-  volatile<TP extends object>(fn: (self: Instance<this>) => TP) {
```

Comment:
Changing all of these to be clearer. `object` is generally not what one wants.

![CleanShot 2024-11-09 at 11 02 56@2x](https://github.com/user-attachments/assets/8e0684b7-4baf-4af0-af13-0962ebf0d917)

I think what we want is only the last.

Reply:
I definitely agree here.